src/widgetApp/mainwindow.h

```diff
@@ -16,8 +24,61 @@ namespace widgetApp{
         MainWindow(QWidget *parent = nullptr);
         ~MainWindow();
 
+    private slots:
+
+        void on_CalculateButton_clicked();
+
+        void on_drawButton_clicked();
+
+        void on_hostNumberSpinBox_valueChanged(int value);
+
+        void on_radioButton_clicked(bool checked);
+
+
     private:
+
         Ui::MainWindow *ui;
+
+        int subnetCount;
+        bool isHorizontal;
+
+        IPv4parser parser;
+        Networkv4 *mainNetwork;
+        SubnetsCalculatorV4 *calculator;
+        std::vector<std::shared_ptr<Subnet>> subnets;
+
+        QWidget *addressWidget;
+        QWidget *maskWidget;
+        QWidget *binaryAddressWidget;
+        QWidget *binaryMaskWidget;
+
+        QSpinBox *subnetCountBox;
+        QWidget *subnetScrollContent;
+        QWidget *subnetGraphContent;
+        QWidget *infoScrollContent;
+
+        QFrame *subnetsGraphFrame;
+
+        QVBoxLayout *subnetsPanelLayout = new QVBoxLayout();
+        QVBoxLayout *infoPanelLayout = new QVBoxLayout();
+        QHBoxLayout *graphPanelLayout = new QHBoxLayout();
+
+        QList<QSpinBox*> *spinBoxList;
```

Comment:
Jeżeli Qt w jakiś sposób nie zarządza pamięcią, to trzeba pozbyć się tych wycieków pamięci.

Reply:
Jeżeli uważnie pisane to nie powinno być leaków:
- https://stackoverflow.com/questions/33508162/does-this-code-provide-memory-leaks
- https://stackoverflow.com/a/33508464/9889714

Reply:
```suggestion
        QList<QSpinBox*> spinBoxList;
```